content/en/blog/_posts/2022-04-15-grpc-probes-in-beta.md

```diff
@@ -0,0 +1,175 @@
+instantiating new processes.
+
+There are a few limitations though. Since configuring a client certificate for probes is hard,
+services that require client authentication are not supported. The built-in probes are also
+not checking the server certificates and ignore related problems.
+
+Built-in checks also cannot be configured to ignore certain types of errors
+(`grpc_health_probe` returns different exit codes for different errors),
+and cannot be "chained" to run the health check on multiple services in a single probe.
+
+But all these limitations are quite standard for gRPC and there are easy workarounds
+for those.
+
+## Try it for yourself
+
+### Getting cluster
+
+You can try this feature today. To try the feature you can spin up the kubernetes cluster
+yourself with the `GRPCContainerProbe` feature gate enabled.
+
+Alternatively, at the moment of writing, you can spin up the cluster on GKE
+using the following command (note, version is `1.23` and `enable-kubernetes-alpha` are specified):
+
+``` bsh
+$ gcloud container clusters create test-grpc \
+    --enable-kubernetes-alpha \
+    --no-enable-autorepair \
+    --no-enable-autoupgrade \
+    --release-channel=rapid \
+    --cluster-version=1.23
+```
+
+You will also need to configure `kubectl` to access the cluster:
+
+``` bsh
+$ gcloud container clusters get-credentials test-grpc
+```
+
+### Trying the feature out
+
+Lets create the pod to test how gRPC probes work. The `agnhost` image has
+a useful [grpc-health-checking](https://github.com/kubernetes/kubernetes/blob/b2c5bd2a278288b5ef19e25bf7413ecb872577a4/test/images/agnhost/README.md#grpc-health-checking) module
+that exposes two ports - one is serving health checking service,
+another - http port to react on commands `make-serving` and `make-not-serving`.
+
+Here is an example pod definition. It starts the `grpc-health-checking` module,
+exposes ports `5000` and `8080`, and configures gRPC readiness probe:
+
+``` yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-grpc
+spec:
+  containers:
+  - name: agnhost
+    image: k8s.gcr.io/e2e-test-images/agnhost:2.35
+    command: ["/agnhost", "grpc-health-checking"]
+    ports:
+    - containerPort: 5000
+    - containerPort: 8080
+    readinessProbe:
+      grpc:
+        port: 5000
+```
+
+If the file called `test.yaml`, you can create the pod and check it's status.
+The pod will be in ready state as indicated by the snippet of the output.
+
+``` bsh
+$ kubectl apply -f test.yaml
+$ kubectl describe test-grpc
+
+...
+Conditions:
+  Type              Status
+  Initialized       True
+  Ready             True
+  ContainersReady   True
+  PodScheduled      True
+...
+```
+
+Now let's change the health checking endpoint status to NOT_SERVING.
+In order to call the http port of the Pod, let's create a port forward:
+
+``` bsh
+$ kubectl port-forward test-grpc 8080:8080
+```
+
+You can `curl` to call the command...
+
+``` bsh
+$ curl http://localhost:8080/make-not-serving
```

Comment:
Don't include a prompt; it makes the code not be valid shell.
Also, the highlight for POSIX shell is `shell` not `bsh`.